### PR TITLE
Use Android linter from cmdline-tools

### PR DESCRIPTION
**Pull request by blasten**
Use the Android linter from cmdline-tools. 
We are currently using the one from `tools/bin/lint` which is very old, and doesn't work with Java 11.

### CODE REVIEW

File: tools/android_lint/bin/main.dart [hunk 1]
```diff
@@ -106,14 +104,13 @@ Future<int> runLint(ArgParser argParser, ArgResults argResults) async {
   if (html) {
     lintArgs.addAll(<String>['--html', argResults['out'] as String]);
   }
-  final String? javaHome = await getJavaHome();
+  final String javahome = getJavaHome(inArgument);
+  print('Using JAVA_HOME=$javahome');
```
Comment (GaryQian) [inline tools/android_lint/bin/main.dart:108]: nit: Is this print meant to be here?

File: shell/platform/android/io/flutter/embedding/android/FlutterImageView.java [hunk 2]
```diff
@@ -99,6 +99,7 @@ private static void logW(String format, Object... args) {
   }
 
   @TargetApi(19)
+  @SuppressLint("WrongConstant") // RGBA_8888 is a valid constant.
```
Comment (GaryQian) [inline shell/platform/android/io/flutter/embedding/android/FlutterImageView.java:102]: Looks like you are forgetting to import `android.annotation.SuppressLint`

File: shell/platform/android/io/flutter/app/FlutterActivityDelegate.java [hunk 3]
```diff
@@ -43,7 +43,7 @@
  * The most obvious example of when this may come in handy is if an application wishes to subclass
  * the Android v4 support library's {@code FragmentActivity}.
  *
- * <p><b>Usage:</b></p>
```
Comment (GaryQian) [inline shell/platform/android/io/flutter/app/FlutterActivityDelegate.java:46]: Any reason why the closing </p> was removed?

**Review by GaryQian — APPROVED** [review]
A few nits, LGTM

File: tools/android_lint/bin/main.dart (hunk 1 above)
Comment (blasten) [reply tools/android_lint/bin/main.dart:108]: it's intended. It helps inform the user of this script that this ENV variable is set. Particularly, relevant in CI.

File: shell/platform/android/io/flutter/embedding/android/FlutterImageView.java (hunk 2 above)
Comment (blasten) [reply shell/platform/android/io/flutter/embedding/android/FlutterImageView.java:103]: 🤦‍♂️ done

File: shell/platform/android/io/flutter/app/FlutterActivityDelegate.java (hunk 3 above)
Comment (blasten) [reply shell/platform/android/io/flutter/app/FlutterActivityDelegate.java:46]: The closing tag `</p>` isn't required.  See this thread https://github.com/google/google-java-format/issues/61

**Pull request merged**